core/node/external_proof_integration_api/src/processor.rs

```diff
@@ -39,20 +42,55 @@ impl Processor {
         }
     }
 
+    pub(crate) async fn verify_proof(
+        &self,
+        Path(l1_batch_number): Path<u32>,
+        Json(payload): Json<VerifyProofRequest>,
+    ) -> Result<(), ProcessorError> {
+        let mut guard = MethodCallGuard::new(MethodType::VerifyProof);
+
+        let l1_batch_number = L1BatchNumber(l1_batch_number);
+        tracing::info!(
+            "Received request to verify proof for batch: {:?}",
+            l1_batch_number
+        );
+
+        let serialized_proof = bincode::serialize(&payload.0)?;
+        let expected_proof = bincode::serialize(
+            &self
+                .blob_store
+                .get::<L1BatchProofForL1>((l1_batch_number, payload.0.protocol_version))
+                .await?,
+        )?;
+
+        if serialized_proof != expected_proof {
+            return Err(ProcessorError::InvalidProof);
+        }
+
+        guard.mark_successful();
```

Comment:
I think this approach is somewhat error-prone. Maybe a more generic way to do it is to implement a tower middleware via layer_fn (see [example](https://docs.rs/tower/latest/tower/layer/fn.layer_fn.html)). This would probably also let you add the "unknown" method label.

Reply:
UPD: will be done as follow-up due to tight timelines.